source/interprocedural_analyses/taint/modelParser.ml

```diff
@@ -3220,52 +3284,122 @@ let rec parse_statement
-                find_clause;
-                where_clause;
-                model_clause;
-                expected_models_clause;
-                unexpected_models_clause;
-              } ->
-      parse_find_clause ~path find_clause
-      |> as_result_error_list
-      >>= fun find ->
-      parse_where_clause ~path ~find_clause:find where_clause
-      |> as_result_error_list
-      >>= fun where ->
-      parse_model_clause
-        ~path
-        ~taint_configuration
-        ~find_clause:find
-        ~is_object_target:(not (ModelQuery.Find.is_callable find))
-        model_clause
-      >>= check_write_to_cache_models ~path ~location ~where
-      |> as_result_error_list
-      >>= fun models ->
-      parse_output_models_clause ~name ~path expected_models_clause
-      >>= fun expected_models ->
-      parse_output_models_clause ~name ~path unexpected_models_clause
-      >>| fun unexpected_models ->
-      [
-        ParsedQuery
-          {
-            ModelQuery.find;
-            where;
-            models;
-            name;
-            logging_group_name;
-            path;
-            location;
-            expected_models;
-            unexpected_models;
-          };
-      ]
+      let result =
+        ModelQueryArguments.parse_arguments ~path ~location arguments
+        |> as_result_error_list
+        >>= fun {
+                  ModelQueryArguments.name;
+                  logging_group_name;
+                  find_clause;
+                  where_clause;
+                  model_clause;
+                  expected_models_clause;
+                  unexpected_models_clause;
+                } ->
+        parse_find_clause ~path find_clause
+        |> as_result_error_list
+        >>= fun find ->
+        parse_where_clause ~path ~find_clause:find where_clause
+        |> as_result_error_list
+        >>= fun where ->
+        parse_model_clause
+          ~path
+          ~taint_configuration
+          ~find_clause:find
+          ~is_object_target:(not (ModelQuery.Find.is_callable find))
+          model_clause
+        >>= check_write_to_cache_models ~path ~location ~where
+        |> as_result_error_list
+        >>= fun models ->
+        parse_output_models_clause ~name ~path expected_models_clause
+        >>= fun expected_models ->
+        parse_output_models_clause ~name ~path unexpected_models_clause
+        >>| fun unexpected_models ->
+        [
+          ParsedQuery
+            {
+              ModelQuery.find;
+              where;
+              models;
+              name;
+              logging_group_name;
+              path;
+              location;
+              expected_models;
+              unexpected_models;
+            };
+        ]
+      in
+      match result with
+      | Ok parsed_statements_list ->
+          List.map ~f:(fun statement -> Ok statement) parsed_statements_list
+      | Error errors_list -> List.map ~f:(fun error -> Error error) errors_list)
```

Comment:
Not a fan of the big `let result = ... in match result ..`. It's a bit hard to read since it forces me to backtrace.
I think we can do better.
First of, we can result directly `Ok (ParsedQuery ...)` line 3319.
For the error case, we might just need to update that `as_result_error_list` function.
If we need further transformation, you can always add a `Result.map_error` at the end.